src/ssl.c

```diff
@@ -8295,6 +8293,31 @@ WOLFSSL_EVP_PKEY* wolfSSL_d2i_PrivateKey(int type, WOLFSSL_EVP_PKEY** out,
     return local;
 }
 
+WOLFSSL_EVP_PKEY* wolfSSL_d2i_PublicKey(int type, WOLFSSL_EVP_PKEY** out,
+        const unsigned char **in, long inSz)
+{
+    WOLFSSL_ENTER("wolfSSL_d2i_PublicKey");
+
+    return wolfSSL_d2i_PublicKey_int(type, out, in, inSz, 0);
+}
+/* Reads in a DER format key. If PKCS8 headers are found they are stripped off.
+ *
+ * type  type of key
+ * out   newly created WOLFSSL_EVP_PKEY structure
+ * in    pointer to input key DER
+ * inSz  size of in buffer
+ *
+ * On success a non null pointer is returned and the pointer in is advanced the
+ * same number of bytes read.
+ */
+WOLFSSL_EVP_PKEY* wolfSSL_d2i_PrivateKey(int type, WOLFSSL_EVP_PKEY** out,
```

Comment:
Body of this function was moved into `wolfSSL_d2i_PublicKey_int` and adapted to support both public and private keys. Now, this function and `wolfSSL_d2i_PublicKey` just wrap `wolfSSL_d2i_PublicKey_int`.